src/chips/overflow_check_v2.rs

```diff
@@ -0,0 +1,120 @@
+use std::fmt::Debug;
+
+use super::utils::{
+    range_check_vec,
+};
+use halo2_proofs::{circuit::*, halo2curves::pasta::Fp, plonk::*, poly::Rotation};
+
+#[derive(Debug, Clone)]
+pub struct OverflowCheckV2Config<const MAX_BITS: u8, const ACC_COLS: usize> {
+    pub value: Column<Advice>,
+    pub decomposed_values: [Column<Advice>; ACC_COLS],
+    pub instance: Column<Instance>,
+    pub selector: Selector,
+}
+
+#[derive(Debug, Clone)]
+pub struct OverflowChipV2<const MAX_BITS: u8, const ACC_COLS: usize> {
+    config: OverflowCheckV2Config<MAX_BITS, ACC_COLS>,
+}
+
+impl<const MAX_BITS: u8, const ACC_COLS: usize> OverflowChipV2<MAX_BITS, ACC_COLS> {
+    pub fn construct(config: OverflowCheckV2Config<MAX_BITS, ACC_COLS>) -> Self {
+        Self { config }
+    }
+
+    pub fn configure(
+        meta: &mut ConstraintSystem<Fp>,
+        value: Column<Advice>,
+        decomposed_values: [Column<Advice>; ACC_COLS],
+        instance: Column<Instance>,
+        selector: Selector,
+    ) -> OverflowCheckV2Config<MAX_BITS, ACC_COLS> {
+        meta.enable_equality(value);
+        decomposed_values.map(|col| meta.enable_equality(col));
+
+        meta.create_gate("range check decomposed values", |meta| {
+            let s_doc = meta.query_selector(selector);
+
+            let value = meta.query_advice(value, Rotation::cur());
+
+            let decomposed_value_vec = (0..ACC_COLS)
+                .map(|i| meta.query_advice(decomposed_values[i], Rotation::cur()))
+                .collect::<Vec<_>>();
+
+            let decomposed_value_sum =
+                (0..=ACC_COLS - 2).fold(decomposed_value_vec[ACC_COLS - 1].clone(), |acc, i| {
+                    acc + (decomposed_value_vec[i].clone()
+                        * Expression::Constant(Fp::from(
+                            1 << (MAX_BITS as usize * ((ACC_COLS - 1) - i)),
+                        )))
+                });
+
+            [
+                vec![s_doc.clone() * (decomposed_value_sum - value)], // equality check between decomposed value and value
+                range_check_vec(&s_doc, decomposed_value_vec, 1 << MAX_BITS), // range check for each decomposed columns
```

Comment:
Can you explain how the range check is performed here? Why did you choose not to use lookup tables?

Reply:
> Can you explain how the range check is performed here? 

The range check is performed by verifying that every value of decomposed_values[i] is within the range of 1 << MAX_BITS.
The main reason for this is that the assigned values in the decomposed columns should not exceed MAX_BITS. In other words, any value greater than MAX_BITS should be carried over and added to the next (left) column.

> Why did you choose not to use lookup tables?

I was not familiar with using the lookup table at that time. Now I tried range check with lookup table [here - sifnoc/overflow-check-range](https://github.com/sifnoc/halo2-experiments/commit/4feed75e724449fec3f884f4d089ddb1d1bc746c) but Error with this massage 
```
thread 'circuits::overflow_check_v2::tests::test_none_overflow_case' panicked at 'called `Result::unwrap()` on an `Err` value: Synthesis'
```
Will try later

Reply:
I can take care of that, if you allow me to do that.

Reply:
Sure! I will be glad

Reply:
added support ce39211 (and further fixing cdacf94)